samples/android/app/src/androidTest/java/org/mozilla/samples/gleancore/pings/SharedTestUtils.kt

```diff
@@ -0,0 +1,36 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.samples.gleancore.pings
+
+import org.json.JSONObject
+import org.mozilla.samples.gleancore.getPingServer
+import java.util.concurrent.TimeUnit
+
+/**
+ * Waits for ping with the given name to be received
+ * in the test ping server.
+ *
+ * @param pingName the name of the ping to wait for
+ * @param maxAttempts the maximum number of attempts
+ */
+fun waitForPingContent(
+    pingName: String,
+    maxAttempts: Int = 3
+): JSONObject?
+{
+    val server = getPingServer()
+
+    var attempts = 0
+    do {
+        attempts += 1
+        val request = server.takeRequest(20L, TimeUnit.SECONDS)
```

Comment:
20 seconds seems like a long time for awaiting networking from a test.

Reply:
That's the standard time we've been using throughout the whole codebase: if the ping reaches the server in 100ms, it stops waiting. However, CI can be slow, and waits can be > 5 seconds.